### PR TITLE
OSD-5226: Improve denial messages for OSD customers

### PR DESCRIPTION
**Pull request by lisa**
Signed-off-by: Lisa Seelye <lisa@users.noreply.github.com>

### CODE REVIEW

File: pkg/webhooks/group/group.go [hunk 1]
```diff
@@ -114,7 +115,7 @@ func (s *GroupWebhook) authorized(request admissionctl.Request) admissionctl.Res
 			}
 		}
 		log.Info("Denying access", "request", request.AdmissionRequest)
-		ret = admissionctl.Denied("May not access protected group")
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from accessing Red Hat managed groups. Customer user groups may NOT match this regular expression: %s", protectedGroups))
```
Comment (jewzaam) [inline pkg/webhooks/group/group.go:118]: We don't need to include the regex, just that they're not permitted.

File: pkg/webhooks/namespace/namespace.go [hunk 2]
```diff
@@ -151,7 +151,7 @@ func (s *NamespaceWebhook) authorized(request admissionctl.Request) admissionctl
 			return ret
 		}
 		log.Info("Non-admin access attempt to privileged namespace", "request", request.AdmissionRequest)
-		ret = admissionctl.Denied("Non-admin access attempt to privileged namespace")
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from creating or changing a potentially harmful Namespace. Customer namespaces should NOT match this regular expressions: %s", privilegedNamespace))
```
Comment (jewzaam) [inline pkg/webhooks/namespace/namespace.go:154]: Don't need to include regex.

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: pkg/webhooks/group/group.go (hunk 1 above)
Comment (lisa) [reply pkg/webhooks/group/group.go:118]: The suggestion was to include the list of groups. However, that has the drawback where it's going to be hard to update as OSD matures. This is a proposal to meet halfway.

File: pkg/webhooks/namespace/namespace.go (hunk 2 above)
Comment (lisa) [reply pkg/webhooks/namespace/namespace.go:154]: As above. https://github.com/openshift/managed-cluster-validating-webhooks/pull/111#discussion_r507714473

**Comment by lisa** [comment]
> @lisa: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> ci/prow/images	[d572b73](https://github.com/openshift/managed-cluster-validating-webhooks/commit/d572b737638b6e24f89397d807078a741404a388)	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_managed-cluster-validating-webhooks/111/pull-ci-openshift-managed-cluster-validating-webhooks-master-images/1318171252480806912)	`/test images`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openshift&repo=managed-cluster-validating-webhooks&pr=111). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:lisa).

#112 will resolve this

**Pull request merged**